folium/map.py

```diff
@@ -297,6 +296,27 @@ def render(self, **kwargs):
             '</style>'), name='map_style')
 
         super(LegacyMap, self).render(**kwargs)
+    
+    @property
+    def location(self):
+        return self._location
```

Comment:
I am not sure we need a property and a setter for this. Can you explain what you have in mind?

I thought that we could have an internal function to check and raise a `ValueError` that we would call in the places we need to check for the two value list.